### DISCUSSION TITLE
Deprecation of DotNext.Reflection

### DISCUSSION THREAD

**Discussion by sakno** [dotnet/dotNext#142] [General]
`DotNext.Reflection` is a library for fast Reflection. .NET Team did amazing work to improve performance of Reflection in the latest release. Therefore, I decided to abandon development of this library. **EOL is December 2023**

Here is the links that can help you to migrate from this library:
* [Performance improvements in .NET 7: Reflection](https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/#reflection). Starting from .NET 7, Reflection utilizes dynamic code generation just like as `DotNext.Reflection`
* [MethodInfo.CreateDelegate&lt;T&gt;](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.methodinfo.createdelegate?view=net-6.0#system-reflection-methodinfo-createdelegate-1) method was added in .NET 5. It allows to create typed delegate for a particular method and use it for invocation. Invocation through delegate has no (or very very small) overhead.
* [UnsafeAccessorAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute) is a new approach introduced in .NET 8 to do reflection call at zero cost.

**Comment by zacharylayne** [comment]
Just curiously, is there anything in the library that will be migrated to one of the other DotNext libraries?

**Reply by sakno** [reply]
I don't think so. But I'm open to discussion, if you see any reason for that.

**Comment by sgf** [comment]
But the .net team's optimization relies on the release of new versions.
Therefore, there is almost no optimization for .net6.
Between this, I think I will still choose to use DotNext.Reflection, even if DotNext.Reflection stop development .
